lib/Parse/ParseStmt.cpp

```diff
@@ -1833,6 +1833,67 @@ Parser::parseStmtConditionElement(SmallVectorImpl<StmtConditionElement> &result,
   return Status;
 }
 
+/// Returns `true` if the current token represents the start of a conditional statement body.
+bool Parser::isStartOfConditionalStmtBody() {
+  Parser::BacktrackingScope Backtrack(*this);
+  
+  if (!Tok.is(tok::l_brace)) {
+    // Statement bodies always start with a '{'. If there is no '{', we can't be at the statement body.
+    return false;
+  }
+  
+  skipSingle();
+  
+  if (Tok.is(tok::eof)) {
+    // There's nothing else in the source file that could be the statement body, so this must be it.
+    return true;
+  }
+  
+  if (Tok.is(tok::semi)) {
+    // We can't have a semicolon between the condition and the statement body, so this must be the statement body.
+    return true;
+  }
+  
+  if (Tok.is(tok::kw_else)) {
+    // If the current token is an `else` keyword, this must be the statement body of an `if` statement since conditions can't be followed by `else`.
+    return true;
+  }
+
+  if (Tok.is(tok::l_paren)) {
+    return false;
+  }
+  
+  if (Tok.is(tok::r_brace) || Tok.is(tok::r_paren)) {
+    // A right brace or parenthesis cannot start a statement body, nor can the condition list continue afterwards. So, this must be the statement body.
+    // This covers cases like `if true, { if true, { } }` or `( if true, { print(0) } )`. While the latter is not valid code, it improves diagnostics.
+    return true;
+  }
+  
+  if (Tok.isAtStartOfLine()) {
+    // If the current token is at the start of a line, it is most likely a statement body. The only exceptions are:
+    if (Tok.is(tok::comma)) {
+      // If newline begins with ',' it must be a condition trailing comma, so this can't be the statement body, e.g.
+      // if true, { true }
+      // , true { print("body") }
+      return false;
+    }
+    if (Tok.is(tok::oper_binary_spaced)) {
+      // If current token is a binary operator this can't be the statement body since an `if` expression can't be the left-hand side of an operator, e.g.
```

Comment:
For readability, you may want to wrap some of these longer comments: 
```cpp
// If current token is a binary operator, this can't be the statement body.
// An `if` expression can't be the left-hand side of an operator, e.g.
```